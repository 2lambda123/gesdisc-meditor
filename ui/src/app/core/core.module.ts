import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
