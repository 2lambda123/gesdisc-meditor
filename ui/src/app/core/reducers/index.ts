import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
