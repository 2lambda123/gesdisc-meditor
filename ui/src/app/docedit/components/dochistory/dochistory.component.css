.history-item {
	margin-top: 5px;
}

.history-item:hover {
	background-color: lightgreen;
	cursor: pointer;
}
