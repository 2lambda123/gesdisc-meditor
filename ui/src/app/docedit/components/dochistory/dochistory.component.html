<mat-card>
	<h3>History</h3>
	<mat-card-content>
		<div *ngFor="let historyItem of dochistory" (click)="loadVersion.emit(historyItem.modifiedOn)">
			<mat-card class="history-item" [ngClass]="[historyItem.modifiedOn == selectedHistory ? 'history-item-selected' : '']">
				<div class="modified-on-label">{{historyItem.modifiedOn | date:'medium' }}</div>
				<div class="modified-by-label">by {{historyItem.modifiedBy}}</div>
			</mat-card>
		</div>
		<div *ngIf="!dochistory.length">Nothing here yet...</div>
	</mat-card-content>
</mat-card>
