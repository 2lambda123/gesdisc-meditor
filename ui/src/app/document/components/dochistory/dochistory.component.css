h3 {
	padding: 24px 24px 0 24px;
	font-size: 1.5em;
	font-weight: bold;
}

.history-content-container {
	padding: 0 24px 24px 24px;
	overflow-y: scroll;
}

.history-item {
	margin-top: 5px;
	padding: 15px;
}

.history-item:hover {
	background-color: #c0efb8;
	cursor: pointer;
}

.history-item-selected {
	background-color: #97b8ed;
}

.history-item-selected:hover {
	background-color: #97b8ed;
}

.modified-on-label {
	text-decoration: underline;
	text-decoration-color: blue;
	color: blue;
}
