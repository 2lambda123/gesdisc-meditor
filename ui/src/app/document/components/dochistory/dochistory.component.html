<div class="history-header">
  <h3>History</h3>
  <button mat-icon-button class="close-icon" (click)="close()">
    <mat-icon>close</mat-icon>
  </button>
</div>
<cdk-virtual-scroll-viewport itemSize="50" class="history-content-container">
  <div *cdkVirtualFor="let historyItem of dochistory" (click)="loadVersion.emit(historyItem.modifiedOn)">
    <mat-card class="history-item" [ngClass]="[historyItem.modifiedOn == selectedHistory ? 'history-item-selected' : '']">
      <div class="modified-on-label">{{historyItem.modifiedOn | date:'medium' }}</div>
      <div class="modified-by-label">by {{historyItem.modifiedBy}}</div>
    </mat-card>
  </div>
</cdk-virtual-scroll-viewport>
<div *ngIf="!dochistory.length">Nothing here yet...</div>
