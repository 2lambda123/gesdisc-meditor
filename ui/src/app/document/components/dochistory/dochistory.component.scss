@import "~main-theme.scss";

h3 {
	font-size: 1.5em;
	font-weight: bold;
}

.history-item {
	margin-top: 5px;
	padding: 15px;
	transition: background .15s ease;
}

.history-item:hover {
	background-color: mat-color($main-theme-primary);
	cursor: pointer;
}

.history-item-selected,
.history-item-selected:hover {
	background-color: mat-color($main-theme-accent, 200);
}

.modified-on-label {
	text-decoration: underline;
	text-decoration-color: blue;
	color: blue;
}
