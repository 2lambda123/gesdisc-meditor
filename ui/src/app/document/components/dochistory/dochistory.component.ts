import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { DocHistory } from '../../../service/model/docHistory';

@Component({
  selector: 'med-doc-history',
  templateUrl: './dochistory.component.html',
  styleUrls: ['./dochistory.component.scss']
})
export class DochistoryComponent implements OnInit {

	@Input() dochistory: DocHistory[];
	@Input() selectedHistory: string;
	@Output() loadVersion = new EventEmitter<string>();
  @Output() closeSidenav: EventEmitter<any> = new EventEmitter();

  constructor() { }

  ngOnInit() {
  }

  close() {
    this.closeSidenav.emit();
  }

}
