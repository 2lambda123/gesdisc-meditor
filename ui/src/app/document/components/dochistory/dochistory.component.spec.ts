import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatCardModule, MatIconModule } from '@angular/material';
import { ScrollingModule } from '@angular/cdk/scrolling';

import { DochistoryComponent } from './dochistory.component';

describe('DochistoryComponent', () => {
  let component: DochistoryComponent;
  let fixture: ComponentFixture<DochistoryComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ 
        NoopAnimationsModule,
        FormsModule,
        ScrollingModule,
        MatCardModule,
        MatIconModule
      ],
      declarations: [ DochistoryComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DochistoryComponent);
    component = fixture.componentInstance;
  });

  it('should create version list with mock data', () => {
    component.selectedHistory = '2018-10-29T18:00:09.424Z'
    
    component.dochistory = [
      {
        modifiedOn: new Date('2018-10-29T18:00:09.424Z'),
        modifiedBy: 'azasorin'
      },
      {
        modifiedOn: new Date('2018-10-29T18:00:04.422Z'),
        modifiedBy: 'azasorin'
      },
      {
        modifiedOn: new Date('2018-10-29T18:00:01.380Z'),
        modifiedBy: 'azasorin'
      }
    ]
    
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
