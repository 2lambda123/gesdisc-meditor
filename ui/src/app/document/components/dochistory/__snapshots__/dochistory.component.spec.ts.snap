// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DochistoryComponent should create version list with mock data 1`] = `
<med-doc-history
  dochistory={[Function Array]}
  loadVersion={[Function EventEmitter]}
  selectedHistory={[Function String]}
>
  <h3>
    History
  </h3><div
    class="history-content-container"
  >
    
    <div>
      <mat-card
        class="history-item mat-card"
        ng-reflect-klass="history-item"
        ng-reflect-ng-class=""
      >
        <div
          class="modified-on-label"
        >
          Oct 29, 2018, 2:00:09 PM
        </div>
        <div
          class="modified-by-label"
        >
          by azasorin
        </div>
      </mat-card>
    </div>
    <div>
      <mat-card
        class="history-item mat-card"
        ng-reflect-klass="history-item"
        ng-reflect-ng-class=""
      >
        <div
          class="modified-on-label"
        >
          Oct 29, 2018, 2:00:04 PM
        </div>
        <div
          class="modified-by-label"
        >
          by azasorin
        </div>
      </mat-card>
    </div>
    <div>
      <mat-card
        class="history-item mat-card"
        ng-reflect-klass="history-item"
        ng-reflect-ng-class=""
      >
        <div
          class="modified-on-label"
        >
          Oct 29, 2018, 2:00:01 PM
        </div>
        <div
          class="modified-by-label"
        >
          by azasorin
        </div>
      </mat-card>
    </div>
    
  </div>
</med-doc-history>
`;
